apps/aechannel/src/aesc_fsm.erl

```diff
@@ -1005,26 +1067,74 @@ withdraw_locked_complete(SignedTx, Updates, #data{state = State, opts = Opts} =
                                                             OnChainEnv, Opts)},
     next_state(open, D1).
 
+settle_signed(SignedTx, Updates, #data{ on_chain_id = ChId} = D) ->
+    lager:debug("SignedTx = ~p", [SignedTx]),
+    case aec_chain:get_channel(ChId) of
+        {ok, Ch} ->
+            LockedUntil = aesc_channels:locked_until(Ch),
+            {ok, D1} =
+                case is_channel_locked(LockedUntil) of
+                    true ->
+                        lager:debug("channel is locked", []),
+                        start_min_depth_watcher(unlock, SignedTx, Updates, D);
+                    false ->
+                        lager:debug("channel not locked, pushing settle", []),
+                        ok = aec_tx_pool:push(SignedTx),
+                        start_min_depth_watcher(close, SignedTx, Updates, D)
+                end,
+            next_state(channel_closing, D1);
+        {error,_} = Error ->
+            close(Error, D)
+    end.
+
+close_solo_signed(SignedTx, _Updates, #data{ on_chain_id = ChId } = D) ->
+    lager:debug("SignedTx = ~p", [SignedTx]),
+    case aec_chain:get_channel(ChId) of
+        {ok, Ch} ->
+            LockedUntil = aesc_channels:locked_until(Ch),
+            case is_channel_locked(LockedUntil) of
+                false ->
+                    lager:debug("channel not locked: solo-closing", []),
+                    case aec_tx_pool:push(SignedTx) of
+                        ok ->
+                            ok;
+                        {error, Reason} ->
+                            lager:debug("Close_solo tx failed: ~p", [Reason]),
+                            ok
+                    end;
+                true ->
+                    lager:debug("channel is locked: cannot solo-close", []),
+                    ok
+            end,
+            %% Let the watcher pick up the tx once it makes it onto the chain
```

Comment:
Where is watcher looking for close? 🤔 

I see the only point where a watcher looks for close `start_min_depth_watcher(close, ...)` is in `settle_signed` that seems not this code path.

Maybe it is implicit in a `aesc_fsm_min_depth_watcher` started previously?

Reply:
Can this be made more explicit by:
* Either enhancing the comment to where this watcher was started and where it was tasked with watching in relation to this code path;
* Or an assertion like "assert that watcher is aware of watching what is relevant for this point, fail otherwise"?

Reply:
@uwiger Thanks for having clarified that the watch request is not necessarily removed from the watcher, and that the watcher reacts to events from chain.

Not necessary to include in PR though I comment here for tracking.
```suggestion
            %% Let the watcher pick up the tx once it makes it onto the chain.
            %% The watcher was started when opening the channel.
            %% The watcher on initialization:
            %% * Subscribed to chain events.
            %% * Stored a never-done "watch" request - processed at each event and calling back to this FSM.
```